sdk/core/Azure.Core/src/Shared/Argument.cs

```diff
@@ -212,5 +212,24 @@ public static string CheckNotNullOrEmpty(string value, string name)
             AssertNotNullOrEmpty(value, name);
             return value;
         }
+
+        public static void AssertHasOnlySupportedHeaders(RequestConditions? requestConditions, string paramName, string[] supportedHeaders)
```

Comment:
Where would this be used? It seems very scenario specific.

Reply:
yes, We will collapse all the request condition headers(If-Match, if-Not-Match, if-Modified-since, if-not-modified-since)  as one parameter, but we need to check if the collapsed parameter 'RequestConditions requestConditions' only set the value to the headers which the service supported. If the value set to unsupported condition header, it will cause unexpected problem. Since it is an parameter, add the check in the Argument so that the check will testable and extensible.

Reply:
@christothes, we're making an exception to the "don't validate headers" rule in this case, because we may be exposing a strongly-typed header on `RequestConditions` that the service may not accept.  If the service doesn't follow API Stewardship guidance and return an error when sent an unsupported header, it could feasibly execute the operation, **causing data loss**.  Based on our analysis of service APIs in the azure-rest-api-specs repo, it is incredibly rare that we will generate the code.  But please do see the linked PR if you're interested in learning more -- I suggested a way that I think will be more clear for folks reading the generated code when we do generate these validations.

Reply:
Is this something that needs to be in every library that uses `Argument`, though? The class is partial so you can declare your own and add methods. I did this for Key Vault SDK for very specific conditions.